distribution/tools/launchers/src/main/java/org/elasticsearch/tools/launchers/MachineDependentHeap.java

```diff
@@ -66,15 +66,32 @@ public MachineDependentHeap(SystemMemoryInfo systemMemoryInfo) {
 
         Path config = configDir.resolve(ELASTICSEARCH_YML);
         try (InputStream in = Files.newInputStream(config)) {
-            return determineHeapSettings(in);
+            return determineHeapSettings(in, parseForcedMemoryInBytes(userDefinedJvmOptions));
         }
     }
 
-    List<String> determineHeapSettings(InputStream config) {
+    static Long parseForcedMemoryInBytes(List<String> userDefinedJvmOptions) {
+        String totalMemoryBytesOption = userDefinedJvmOptions.stream()
+            .filter(option -> option.startsWith("-Des.total_memory_bytes="))
+            .findFirst()
+            .orElse(null);
+        if (totalMemoryBytesOption == null) {
+            return null;
+        }
+        try {
+            return Long.parseLong(totalMemoryBytesOption.split("=", 2)[1]);
+        } catch (NumberFormatException e) {
+            throw new IllegalArgumentException("Unable to parse number of bytes from [" + totalMemoryBytesOption + "]");
+        }
```

Comment:
I think you don't need an intermediate variable if you used `Optional.map`?

```suggestion
        return userDefinedJvmOptions.stream()
            .filter(option -> option.startsWith("-Des.total_memory_bytes="))
            .findFirst()
            .map(totalMemoryBytesOption -> {
	            try {
	                return Long.parseLong(totalMemoryBytesOption.split("=", 2)[1]);
	            } catch (NumberFormatException e) {
	                throw new IllegalArgumentException("Unable to parse number of bytes from [" + totalMemoryBytesOption + "]");
	            }
            })
            .orElse(null);
```